src/main/java/com/microfocus/application/automation/tools/octane/actions/UFTTestDetectionPublisher.java

```diff
@@ -77,13 +86,56 @@ public UFTTestDetectionPublisher(String configurationId, String workspaceName, S
 
 	@Override
 	public boolean perform(AbstractBuild build, Launcher launcher, BuildListener listener) {
-		UftTestDiscoveryResult results = UFTTestDetectionService.startScanning(build, getConfigurationId(), getWorkspaceName(), getScmRepositoryId(), listener);
+		if (configurationId == null) {
+			logger.error("discovery configurationId is null.");
+			configurationId = tryFindConfigurationId(build, Long.parseLong(workspaceName));
+			if (configurationId != null) {
+				try {
+					build.getParent().save();
+				} catch (IOException e) {
+					logger.error("Failed to save UFTTestDetectionPublisher : " + e.getMessage());
+				}
+			} else {
+				logger.error("No relevant ALM Octane configuration is found.");
+				throw new IllegalArgumentException("No relevant ALM Octane configuration is found.");
```

Comment:
Is throwing an exception in the perform will stop the job? if so with what build status?